front/src/containers/studentPage.js

```diff
@@ -49,13 +50,96 @@ function StudentPage({exercice}) {
     pages: ['Exo 1', 'Exo 2', 'Exo 3', 'Exo 4'],
   };
   const {user, logoutUser} = useContext(AuthContext);
+
+  const [code, setCode] = useState('#enter your code here!');
+  const [listExercises, setListExercises] = useState([{statement: ''}]);
+  const [activeExerciseIndex, setActiveExerciseIndex] = useState(0);
+  const [activeExerciseStatement, setActiveExerciseStatement]= useState('');
```

Comment:
states should not be out of a component.
If needed for a future refactor, you can use react components by class.

Reply:
They aren't outside of the component, they are inside the StudentPage component